### PR TITLE
[Schedule] Loop-Partition Scheduling Primitive

### PR DESCRIPTION
**Pull request by rutkoor**
This PR introduces Loop-Partition scheduling primitive to decompose the single loop into sequence of multiple loops.
This PR is extension of previously created PR(https://github.com/apache/tvm/pull/15901)

### CODE REVIEW

File: src/tir/schedule/primitive/get_block_loop.cc [hunk 1]
```diff
@@ -238,7 +238,7 @@ struct GetOutputBlocksTraits : public UnpackedInstTraits<GetOutputBlocksTraits>
 
   static String UnpackedAsPython(Array<String> outputs, String block_rv) {
     PythonAPICall py("get_output_blocks");
-    py.Input("block", block_rv);
+    py.Input("scope_block", block_rv);
```
Comment (Hzfengsy) [inline src/tir/schedule/primitive/get_block_loop.cc:241]: I'm not sure what the purpose of this change. Maybe a typo?

File: src/tir/schedule/concrete_schedule.cc [hunk 2]
```diff
@@ -500,6 +500,143 @@ Array<LoopRV> ConcreteScheduleNode::Split(const LoopRV& loop_rv,
...
+    }
+
+    IRModule mod() const final { return mod_; }
+    Array<ObjectRef> LocationsOfInterest() const final { return {loop_}; }
+
+    IRModule mod_;
+    For loop_;
+  };
+
+  class NonPositiveFactorError : public ScheduleError {
```
Comment (Hzfengsy) [inline src/tir/schedule/concrete_schedule.cc:564]: Would be great to reuse the same error class as Split.

**Review by Hzfengsy — COMMENTED** [review]
Overall LGTM. Thanks for the great work!

File: src/tir/schedule/primitive/get_block_loop.cc (hunk 1 above)
Comment (rutkoor) [reply src/tir/schedule/primitive/get_block_loop.cc]: Removed.

File: src/tir/schedule/concrete_schedule.cc (hunk 2 above)
Comment (rutkoor) [reply src/tir/schedule/concrete_schedule.cc]: Added changes to reuse the same error class for Split and Loop-Partition.

**Review by Hzfengsy — APPROVED** [review]

**Comment by Hzfengsy** [comment]
Thanks @rutkoor

**Pull request merged**